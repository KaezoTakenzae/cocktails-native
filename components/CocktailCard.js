import React from 'react';
import {
  Image,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  TouchableNativeFeedback,
  Platform
} from 'react-native';

import { Text, View } from 'components/Themed';
import IngredientsList from './IngredientsList';

export default function CocktailCard({data, ingredients, viewDetails}) {
  let TouchableCmp = TouchableOpacity;

  if (Platform.OS === 'android' && Platform.Version >= 21) {
    TouchableCmp = TouchableNativeFeedback;
  }

  let {
    strDrink,
    strDrinkThumb,
    idDrink
  } = data;
  return (
    <View style={styles.root}>
      <TouchableCmp onPress={viewDetails} useForeground>
        <View>
          <View style={styles.mainContent}>
            <Text style={styles.title}>{strDrink}</Text>
            <Image style={styles.image} source={{ uri: strDrinkThumb }} />
          </View>
          {ingredients ? <IngredientsList data={ingredients} shortList /> : null}
        </View>
      </TouchableCmp>
    </View>
  );
}

const styles = StyleSheet.create({
  root: {
    padding: 8,
    backgroundColor: 'white',
    borderBottomWidth: 1,
    borderBottomColor: 'lightgrey'
  },
  mainContent: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  image: {
    height: 50,
    width: 50,
    width: '20%'
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    width: '80%'
  }
});
