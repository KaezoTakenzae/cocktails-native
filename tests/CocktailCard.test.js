import React from 'react';
import renderer from 'react-test-renderer';
import CocktailCard from 'components/CocktailCard';

let cocktail = {
  strDrinkThumb: '',
  strDrink: 'White Russian'
}

function findInChildren(predicate) {
  return testInstance => {
    const children = testInstance.children
    return Array.isArray(children)
      ? children.some(predicate)
      : predicate(children)
  }
}

it(`renders correctly`, () => {
  const tree = renderer.create(
    <CocktailCard
      data={cocktail}
      viewDetails={() => {}}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Drink title is White Russian', () => {
  const tree = renderer.create(
    <CocktailCard
      data={cocktail}
      viewDetails={() => {}}
    />
  );
  const instance = tree.root.findByType('Text');

  expect(instance.props.children).toBe('White Russian');
});

it('Card can be clicked and function will run', () => {
  let btnClicked = false;
  const tree = renderer.create(
    <CocktailCard
      data={cocktail}
      viewDetails={() => { btnClicked = true; }}
    />
  );
  const instance = tree.root.findByProps({ focusable: true });
  instance.props.onClick();

  expect(btnClicked).toBe(true);
});
