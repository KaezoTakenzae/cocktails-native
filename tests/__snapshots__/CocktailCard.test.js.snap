// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "lightgrey",
        "borderBottomWidth": 1,
        "padding": 8,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "width": "80%",
              },
            ]
          }
        >
          White Russian
        </Text>
        <Image
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Object {
              "height": 50,
              "width": "20%",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
